modules/venat-module-xiv-market/src/data/listings.ts

```diff
@@ -2,54 +2,61 @@ import { LookupResult } from '@the-convocation/venat-core';
 import { XIVAPIItem } from './xivapi';
 
-export interface MarketListing {
+export interface ItemMarketListing {
   pricePerUnit: number;
   quantity: number;
   total: number;
   hq: boolean;
   worldName?: string;
 }
 
-export interface MarketInfo {
+export interface ItemMarketInfo {
   itemId: number;
   itemName: string;
   worldName?: string;
   dcName?: string;
   lastUploadTime: number;
-  listings: MarketListing[];
+  listings: ItemMarketListing[];
 }
 
+type GetItemInfoByName = (
+  itemName: string,
+) => Promise<LookupResult<XIVAPIItem>>;
+
+type GetItemMarketInfo = (
+  itemId: number,
+  server: string,
+) => Promise<LookupResult<UniversalisMarketInfo>>;
+
+type LogError = (message?: string, stack?: string) => void;
+
+/**
+ * Gets an item's market info using its name and a game server. Requires provider functions for I/O.
+ * @param getItemInfoByName An async function that returns an item's information by its name.
+ * @param getMarketInfo An async function that returns an item's market info by its item ID and a game server.
+ * @param logError An error-logging function.
+ * @returns An async function that returns an item's market info by its name and a game server.
+ */
 export function getMarketInfoByName(
-  getItemIdByName: (itemName: string) => Promise<LookupResult<XIVAPIItem>>,
-  getMarketInfo: (
-    itemId: number,
-    server: string,
-  ) => Promise<LookupResult<UniversalisMarketInfo>>,
-  logError: (message?: string, stack?: string) => void,
-): (itemName: string, server: string) => Promise<MarketInfo | string> {
+  getItemInfoByName: GetItemInfoByName,
+  getMarketInfo: GetItemMarketInfo,
+  logError: LogError,
+): (itemName: string, server: string) => Promise<ItemMarketInfo | string> {
   return async (
     itemName: string,
     server: string,
-  ): Promise<MarketInfo | string> => {
-    const itemLookup = await getItemIdByName(itemName);
+  ): Promise<ItemMarketInfo | string> => {
+    const itemLookup = await getItemInfoByName(itemName);
     if (!itemLookup.success) {
       logError(itemLookup.err.message, itemLookup.err.stack);
-      return 'Failed to access XIVAPI; please try again later.';
-    }
-
-    if (!itemLookup.success) {
-      return 'The item could not be found; please check your spelling and try again.';
+      return 'The item could not be found; please check your item name spelling and try again.';
     }
 
     const item = itemLookup.value;
     const marketLookup = await getMarketInfo(item.ID, server);
     if (!marketLookup.success) {
       logError(marketLookup.err.message, marketLookup.err.stack);
-      return 'The item could not be found; please check your spelling of the server and try again.';
-    }
-
-    if (!marketLookup.success) {
-      return 'The item could not be found; please check your spelling and try again.';
+      return 'The item could not be found on the market; please check your server spelling and try again.';
```

Comment:
Should we have an explicit return if the item searched for is a non-market item and let the user know the item they searched for cannot be sold on the market instead, just in case they are spelling it right but still aren't getting results.

Reply:
That's a good idea, I'll add that.

Reply:
Current version of the bot seems to fail for what it's worth. So I don't think it's being handled at all.
![image](https://user-images.githubusercontent.com/39720490/168634204-01afa748-2f56-485d-ac0c-bae8502af765.png)

Reply:
Added.